### PR TITLE
chore: align sonar code coverage file resolution/ignore with jest for…

### PR DESCRIPTION
**Pull request by themooneer**
… lld & llm

### ✅ Checklist

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

### 📝 Description

While comparing Sonar Code Coverage and Jest LCOV report is not relevant, in this PR i fixed : 
- LCOV Path resolution 
- refine the sonar code coverage by ignoring almost same jest.config files from being scanned for unit-test report
- add merge lcov file as a step in the workflow and get rid of useless directory such as desktop-coverage and mobile-coverage generated previously while */coverage directory is already ignored by SCM. 

As result 
- LLD CC increased from 23,4% to 26,6% as the nearest pct comparing to LCOV line/line coverage. (5 to 10% gap because of additional sonar metrics) 
- LLM CC previously 15,1 to 15.7 

##### Reminder 

1-to-1 comparison between SonarQube and coverage report is not relevant
We sometimes have users reporting that the code coverage is different between SonarQube and the tool used to gather it.
The reason for this is most often because people are not comparing the same metrics.
SonarQube gets the covered lines from the coverage report given to the analyzer. Then it calculates all its coverage metrics from there and the executable lines also called lines to cover.
What is often compared is the Line Coverage, most often displayed by the external tool used to gather the covered lines, and what we define as Code Coverage, which is computed from the numbers extracted from the coverage report passed to the analyzer.

### ❓ Context

- **JIRA or GitHub link**: 
https://ledgerhq.atlassian.net/browse/LIVE-14731

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: .github/workflows/sonar.yml [hunk 1]
```diff
@@ -52,8 +50,16 @@ jobs:
 
       - name: Generate Unit test coverage for LLD & LLM
         run: |
-          pnpm desktop test:jest:coverage --coverageDirectory=desktop-coverage
-          pnpm mobile test:jest:coverage --coverageDirectory=mobile-coverage
+          pnpm desktop test:jest:coverage --collectCoverage=true
+          pnpm mobile test:jest:coverage --collectCoverage=true
```
Comment (KVNLS) [inline .github/workflows/sonar.yml:54]: Do we need to use `--collectCoverage=true` as for both commands we use `--coverage` which is an alias to `collectCoverage=true`?

File: apps/ledger-live-desktop/jest.config.js [hunk 2]
```diff
@@ -72,10 +72,14 @@ module.exports = {
     "!src/**/__integration__/**",
     "!src/**/__tests__/**",
   ],
-  coverageReporters: ["json", "lcov", "json-summary"],
+  collectCoverage: true,
```
Comment (KVNLS) [inline apps/ledger-live-desktop/jest.config.js:75]: I think that we might not want to collect coverage as default. When we develop a feature we just want to replay the test quickly. And we can control coverage collection through the cli. Is there a reason to force it in the config?

File: apps/ledger-live-mobile/jest.config.js [hunk 3]
```diff
@@ -47,17 +47,21 @@ module.exports = {
   ],
   testPathIgnorePatterns: ["<rootDir>/node_modules/"],
   moduleDirectories: ["node_modules"],
+  collectCoverage: true,
```
Comment (KVNLS) [inline apps/ledger-live-mobile/jest.config.js:50]: Same for coverage as default, is it needed?

File: apps/ledger-live-desktop/jest.config.js (hunk 2 above)
Comment (themooneer) [reply apps/ledger-live-desktop/jest.config.js:75]: yes ! just forgotten these lines

File: apps/ledger-live-mobile/jest.config.js (hunk 3 above)
Comment (themooneer) [reply apps/ledger-live-mobile/jest.config.js:50]: yes sure

File: .github/workflows/sonar.yml (hunk 1 above)
Comment (themooneer) [reply .github/workflows/sonar.yml]: you right, just added it to make sure unit test field in sonar take coverage per file .. just removed thanks

File: apps/ledger-live-mobile/jest.config.js (hunk 3 above)
Comment (themooneer) [reply apps/ledger-live-mobile/jest.config.js]: deleted

**Review by KVNLS — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**